Tasks/Common/MSBuildHelpers/InvokeFunctions.ps1

```diff
@@ -32,9 +32,7 @@ function Invoke-BuildTools {
             if ($Clean) {
                 Invoke-MSBuild -ProjectFile $file -Targets Clean -MSBuildPath $MSBuildLocation -AdditionalArguments $MSBuildArguments -NoTimelineLogger:$NoTimelineLogger @splat
             }
-
-            # If we cleaned and passed /t targets, we don't need to run them again
-            if (!$Clean -or -not $MSBuildArguments.Contains("/t")) {
+            else {
```

Comment:
Typically `/t:Clean` just runs the clean targets, not the build targets.

Removing the else is fine as long as consumers call build twice?